File: kvt-projekat2024/src/app/models/Exercise.ts
import { User } from "./User";
import { Facility } from "./Facility";

export class Exercise {
  id: number;
  fromTime: Date;
  untilTime: Date;
  user: User;
  facility: Facility;

  constructor(id: number, fromTime: Date, untilTime: Date, user: User, facility: Facility) {
    this.id = id;
    this.fromTime = fromTime;
    this.untilTime = untilTime;
    this.user = user;
    this.facility = facility;
  }
}

File: src/main/java/rs/ac/uns/ftn/svt/model/Exercise.java
package rs.ac.uns.ftn.svt.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@JsonIgnoreProperties({"facility", "user"})
public class Exercise {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime fromTime;

    private LocalDateTime untilTime;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "facility_id", nullable = false)
    private Facility facility;

}

